Tsan fail in unit tests `ComputeServerRunner`

- ComputeServerRunner.randomFailpointForPipeline
```
[2023-06-07T18:28:35.572Z] WARNING: ThreadSanitizer: data race (pid=391079)
[2023-06-07T18:28:35.572Z]   Write of size 4 at 0x00000fcbcaf0 by thread T81 (mutexes: read M734368214561294336):
[2023-06-07T18:28:35.572Z]     #0 randd /home/jenkins/agent/workspace/tiflash-sanitizer-daily/tiflash/contrib/client-c/third_party/libfiu/libfiu/fiu.c:212:11 (gtests_dbms+0xf762dd0)
[2023-06-07T18:28:35.572Z]     #1 fiu_fail /home/jenkins/agent/workspace/tiflash-sanitizer-daily/tiflash/contrib/client-c/third_party/libfiu/libfiu/fiu.c:325:32 (gtests_dbms+0xf762dd0)
[2023-06-07T18:28:35.572Z]     #2 DB::EventTask::executeImpl() /home/jenkins/agent/workspace/tiflash-sanitizer-daily/tiflash/dbms/src/Flash/Pipeline/Schedule/Tasks/EventTask.cpp:82:5 (gtests_dbms+0xdeafb53)
[2023-06-07T18:28:35.572Z]     #3 DB::Task::execute() /home/jenkins/agent/workspace/tiflash-sanitizer-daily/tiflash/dbms/src/Flash/Pipeline/Schedule/Tasks/Task.cpp:102:18 (gtests_dbms+0xdeb8049)
[2023-06-07T18:28:35.572Z]     #4 DB::CPUImpl::exec(std::__1::unique_ptr<DB::Task, std::__1::default_delete<DB::Task> >&) /home/jenkins/agent/workspace/tiflash-sanitizer-daily/tiflash/dbms/src/Flash/Pipeline/Schedule/ThreadPool/TaskThreadPoolImpl.h:34:22 (gtests_dbms+0xdec1580)
[2023-06-07T18:28:35.572Z]     #5 DB::TaskThreadPool<DB::CPUImpl>::handleTask(std::__1::unique_ptr<DB::Task, std::__1::default_delete<DB::Task> >&) /home/jenkins/agent/workspace/tiflash-sanitizer-daily/tiflash/dbms/src/Flash/Pipeline/Schedule/ThreadPool/TaskThreadPool.cpp:100:18 (gtests_dbms+0xdec1580)
[2023-06-07T18:28:35.572Z]     #6 DB::TaskThreadPool<DB::CPUImpl>::doLoop(unsigned long) /home/jenkins/agent/workspace/tiflash-sanitizer-daily/tiflash/dbms/src/Flash/Pipeline/Schedule/ThreadPool/TaskThreadPool.cpp:80:9 (gtests_dbms+0xdec1206)
[2023-06-07T18:28:35.572Z]     #7 DB::TaskThreadPool<DB::CPUImpl>::loop(unsigned long) /home/jenkins/agent/workspace/tiflash-sanitizer-daily/tiflash/dbms/src/Flash/Pipeline/Schedule/ThreadPool/TaskThreadPool.cpp:59:9 (gtests_dbms+0xdec0b26)
[2023-06-07T18:28:35.572Z]     #8 decltype(*(static_cast<DB::TaskThreadPool<DB::CPUImpl>*>(fp0)).*fp(static_cast<unsigned long>(fp1))) std::__1::__invoke<void (DB::TaskThreadPool<DB::CPUImpl>::*)(unsigned long), DB::TaskThreadPool<DB::CPUImpl>*, unsigned long, void>(void (DB::TaskThreadPool<DB::CPUImpl>::*&&)(unsigned long), DB::TaskThreadPool<DB::CPUImpl>*&&, unsigned long&&) /usr/local/bin/../include/c++/v1/type_traits:3859:1 (gtests_dbms+0xdec3990)
[2023-06-07T18:28:35.572Z]     #9 void std::__1::__thread_execute<std::__1::unique_ptr<std::__1::__thread_struct, std::__1::default_delete<std::__1::__thread_struct> >, void (DB::TaskThreadPool<DB::CPUImpl>::*)(unsigned long), DB::TaskThreadPool<DB::CPUImpl>*, unsigned long, 2ul, 3ul>(std::__1::tuple<std::__1::unique_ptr<std::__1::__thread_struct, std::__1::default_delete<std::__1::__thread_struct> >, void (DB::TaskThreadPool<DB::CPUImpl>::*)(unsigned long), DB::TaskThreadPool<DB::CPUImpl>*, unsigned long>&, std::__1::__tuple_indices<2ul, 3ul>) /usr/local/bin/../include/c++/v1/thread:280:5 (gtests_dbms+0xdec3990)
[2023-06-07T18:28:35.572Z]     #10 void* std::__1::__thread_proxy<std::__1::tuple<std::__1::unique_ptr<std::__1::__thread_struct, std::__1::default_delete<std::__1::__thread_struct> >, void (DB::TaskThreadPool<DB::CPUImpl>::*)(unsigned long), DB::TaskThreadPool<DB::CPUImpl>*, unsigned long> >(void*) /usr/local/bin/../include/c++/v1/thread:291:5 (gtests_dbms+0xdec3990)
[2023-06-07T18:28:35.572Z] 
[2023-06-07T18:28:35.572Z]   Previous write of size 4 at 0x00000fcbcaf0 by thread T87 (mutexes: read M734368214561294336):
[2023-06-07T18:28:35.572Z]     #0 randd /home/jenkins/agent/workspace/tiflash-sanitizer-daily/tiflash/contrib/client-c/third_party/libfiu/libfiu/fiu.c:212:11 (gtests_dbms+0xf762dd0)
[2023-06-07T18:28:35.572Z]     #1 fiu_fail /home/jenkins/agent/workspace/tiflash-sanitizer-daily/tiflash/contrib/client-c/third_party/libfiu/libfiu/fiu.c:325:32 (gtests_dbms+0xf762dd0)
[2023-06-07T18:28:35.572Z]     #2 DB::EventTask::executeImpl() /home/jenkins/agent/workspace/tiflash-sanitizer-daily/tiflash/dbms/src/Flash/Pipeline/Schedule/Tasks/EventTask.cpp:82:5 (gtests_dbms+0xdeafb53)
[2023-06-07T18:28:35.572Z]     #3 DB::Task::execute() /home/jenkins/agent/workspace/tiflash-sanitizer-daily/tiflash/dbms/src/Flash/Pipeline/Schedule/Tasks/Task.cpp:102:18 (gtests_dbms+0xdeb8049)
[2023-06-07T18:28:35.572Z]     #4 DB::CPUImpl::exec(std::__1::unique_ptr<DB::Task, std::__1::default_delete<DB::Task> >&) /home/jenkins/agent/workspace/tiflash-sanitizer-daily/tiflash/dbms/src/Flash/Pipeline/Schedule/ThreadPool/TaskThreadPoolImpl.h:34:22 (gtests_dbms+0xdec1580)
[2023-06-07T18:28:35.572Z]     #5 DB::TaskThreadPool<DB::CPUImpl>::handleTask(std::__1::unique_ptr<DB::Task, std::__1::default_delete<DB::Task> >&) /home/jenkins/agent/workspace/tiflash-sanitizer-daily/tiflash/dbms/src/Flash/Pipeline/Schedule/ThreadPool/TaskThreadPool.cpp:100:18 (gtests_dbms+0xdec1580)
[2023-06-07T18:28:35.572Z]     #6 DB::TaskThreadPool<DB::CPUImpl>::doLoop(unsigned long) /home/jenkins/agent/workspace/tiflash-sanitizer-daily/tiflash/dbms/src/Flash/Pipeline/Schedule/ThreadPool/TaskThreadPool.cpp:80:9 (gtests_dbms+0xdec1206)
[2023-06-07T18:28:35.572Z]     #7 DB::TaskThreadPool<DB::CPUImpl>::loop(unsigned long) /home/jenkins/agent/workspace/tiflash-sanitizer-daily/tiflash/dbms/src/Flash/Pipeline/Schedule/ThreadPool/TaskThreadPool.cpp:59:9 (gtests_dbms+0xdec0b26)
[2023-06-07T18:28:35.573Z]     #8 decltype(*(static_cast<DB::TaskThreadPool<DB::CPUImpl>*>(fp0)).*fp(static_cast<unsigned long>(fp1))) std::__1::__invoke<void (DB::TaskThreadPool<DB::CPUImpl>::*)(unsigned long), DB::TaskThreadPool<DB::CPUImpl>*, unsigned long, void>(void (DB::TaskThreadPool<DB::CPUImpl>::*&&)(unsigned long), DB::TaskThreadPool<DB::CPUImpl>*&&, unsigned long&&) /usr/local/bin/../include/c++/v1/type_traits:3859:1 (gtests_dbms+0xdec3990)
[2023-06-07T18:28:35.573Z]     #9 void std::__1::__thread_execute<std::__1::unique_ptr<std::__1::__thread_struct, std::__1::default_delete<std::__1::__thread_struct> >, void (DB::TaskThreadPool<DB::CPUImpl>::*)(unsigned long), DB::TaskThreadPool<DB::CPUImpl>*, unsigned long, 2ul, 3ul>(std::__1::tuple<std::__1::unique_ptr<std::__1::__thread_struct, std::__1::default_delete<std::__1::__thread_struct> >, void (DB::TaskThreadPool<DB::CPUImpl>::*)(unsigned long), DB::TaskThreadPool<DB::CPUImpl>*, unsigned long>&, std::__1::__tuple_indices<2ul, 3ul>) /usr/local/bin/../include/c++/v1/thread:280:5 (gtests_dbms+0xdec3990)
[2023-06-07T18:28:35.573Z]     #10 void* std::__1::__thread_proxy<std::__1::tuple<std::__1::unique_ptr<std::__1::__thread_struct, std::__1::default_delete<std::__1::__thread_struct> >, void (DB::TaskThreadPool<DB::CPUImpl>::*)(unsigned long), DB::TaskThreadPool<DB::CPUImpl>*, unsigned long> >(void*) /usr/local/bin/../include/c++/v1/thread:291:5 (gtests_dbms+0xdec3990)
[2023-06-07T18:28:35.573Z] 
[2023-06-07T18:28:35.573Z]   Location is global 'randd_xn' of size 4 at 0x00000fcbcaf0 (gtests_dbms+0x00000fcbcaf0)
[2023-06-07T18:28:35.573Z] 
[2023-06-07T18:28:35.573Z]   Mutex M734368214561294336 is already destroyed.
[2023-06-07T18:28:35.573Z] 
[2023-06-07T18:28:35.573Z]   Thread T81 'cpu_pool' (tid=391243, running) created by main thread at:
[2023-06-07T18:28:35.573Z]     #0 pthread_create /llvm-project/compiler-rt/lib/tsan/rtl/tsan_interceptors_posix.cpp:977:3 (gtests_dbms+0x28198fd)
[2023-06-07T18:28:35.573Z]     #1 std::__1::__libcpp_thread_create(unsigned long*, void* (*)(void*), void*) /usr/local/bin/../include/c++/v1/__threading_support:514:10 (gtests_dbms+0xdec3880)
[2023-06-07T18:28:35.573Z]     #2 std::__1::thread::thread<void (DB::TaskThreadPool<DB::CPUImpl>::*)(unsigned long), DB::TaskThreadPool<DB::CPUImpl>*, unsigned long&, void>(void (DB::TaskThreadPool<DB::CPUImpl>::*&&)(unsigned long), DB::TaskThreadPool<DB::CPUImpl>*&&, unsigned long&) /usr/local/bin/../include/c++/v1/thread:307:16 (gtests_dbms+0xdec3880)
[2023-06-07T18:28:35.573Z]     #3 std::__1::thread* std::__1::construct_at<std::__1::thread, void (DB::TaskThreadPool<DB::CPUImpl>::*)(unsigned long), DB::TaskThreadPool<DB::CPUImpl>*, unsigned long&, std::__1::thread*>(std::__1::thread*, void (DB::TaskThreadPool<DB::CPUImpl>::*&&)(unsigned long), DB::TaskThreadPool<DB::CPUImpl>*&&, unsigned long&) /usr/local/bin/../include/c++/v1/__memory/construct_at.h:37:38 (gtests_dbms+0xdec0329)
[2023-06-07T18:28:35.573Z]     #4 void std::__1::allocator_traits<std::__1::allocator<std::__1::thread> >::construct<std::__1::thread, void (DB::TaskThreadPool<DB::CPUImpl>::*)(unsigned long), DB::TaskThreadPool<DB::CPUImpl>*, unsigned long&, void, void>(std::__1::allocator<std::__1::thread>&, std::__1::thread*, void (DB::TaskThreadPool<DB::CPUImpl>::*&&)(unsigned long), DB::TaskThreadPool<DB::CPUImpl>*&&, unsigned long&) /usr/local/bin/../include/c++/v1/__memory/allocator_traits.h:298:9 (gtests_dbms+0xdec0329)
[2023-06-07T18:28:35.573Z]     #5 void std::__1::vector<std::__1::thread, std::__1::allocator<std::__1::thread> >::__construct_one_at_end<void (DB::TaskThreadPool<DB::CPUImpl>::*)(unsigned long), DB::TaskThreadPool<DB::CPUImpl>*, unsigned long&>(void (DB::TaskThreadPool<DB::CPUImpl>::*&&)(unsigned long), DB::TaskThreadPool<DB::CPUImpl>*&&, unsigned long&) /usr/local/bin/../include/c++/v1/vector:948:5 (gtests_dbms+0xdec0329)
[2023-06-07T18:28:35.573Z]     #6 std::__1::thread& std::__1::vector<std::__1::thread, std::__1::allocator<std::__1::thread> >::emplace_back<void (DB::TaskThreadPool<DB::CPUImpl>::*)(unsigned long), DB::TaskThreadPool<DB::CPUImpl>*, unsigned long&>(void (DB::TaskThreadPool<DB::CPUImpl>::*&&)(unsigned long), DB::TaskThreadPool<DB::CPUImpl>*&&, unsigned long&) /usr/local/bin/../include/c++/v1/vector:1706:9 (gtests_dbms+0xdec0329)
[2023-06-07T18:28:35.573Z]     #7 DB::TaskThreadPool<DB::CPUImpl>::TaskThreadPool(DB::TaskScheduler&, DB::ThreadPoolConfig const&) /home/jenkins/agent/workspace/tiflash-sanitizer-daily/tiflash/dbms/src/Flash/Pipeline/Schedule/ThreadPool/TaskThreadPool.cpp:37:17 (gtests_dbms+0xdec0329)
[2023-06-07T18:28:35.573Z]     #8 DB::TaskScheduler::TaskScheduler(DB::TaskSchedulerConfig const&) /home/jenkins/agent/workspace/tiflash-sanitizer-daily/tiflash/dbms/src/Flash/Pipeline/Schedule/TaskScheduler.cpp:26:7 (gtests_dbms+0xde9d127)
[2023-06-07T18:28:35.573Z]     #9 std::__1::__unique_if<DB::TaskScheduler>::__unique_single std::__1::make_unique<DB::TaskScheduler, DB::TaskSchedulerConfig&>(DB::TaskSchedulerConfig&) /usr/local/bin/../include/c++/v1/__memory/unique_ptr.h:728:32 (gtests_dbms+0x5c0efe7)
[2023-06-07T18:28:35.573Z]     #10 DB::tests::ExecutorTest::SetUp() /home/jenkins/agent/workspace/tiflash-sanitizer-daily/tiflash/dbms/src/TestUtils/ExecutorTestUtils.cpp:65:31 (gtests_dbms+0x5c0efe7)
[2023-06-07T18:28:35.573Z]     #11 void testing::internal::HandleSehExceptionsInMethodIfSupported<testing::Test, void>(testing::Test*, void (testing::Test::*)(), char const*) /home/jenkins/agent/workspace/tiflash-sanitizer-daily/tiflash/contrib/googletest/googletest/src/gtest.cc:2443:10 (gtests_dbms+0xb8250f5)
[2023-06-07T18:28:35.573Z]     #12 void testing::internal::HandleExceptionsInMethodIfSupported<testing::Test, void>(testing::Test*, void (testing::Test::*)(), char const*) /home/jenkins/agent/workspace/tiflash-sanitizer-daily/tiflash/contrib/googletest/googletest/src/gtest.cc:2479:14 (gtests_dbms+0xb8250f5)
[2023-06-07T18:28:35.573Z]     #13 testing::Test::Run() /home/jenkins/agent/workspace/tiflash-sanitizer-daily/tiflash/contrib/googletest/googletest/src/gtest.cc:2513:3 (gtests_dbms+0xb7fe9a4)
[2023-06-07T18:28:35.573Z]     #14 testing::TestInfo::Run() /home/jenkins/agent/workspace/tiflash-sanitizer-daily/tiflash/contrib/googletest/googletest/src/gtest.cc:2693:11 (gtests_dbms+0xb8004c4)
[2023-06-07T18:28:35.573Z]     #15 testing::TestCase::Run() /home/jenkins/agent/workspace/tiflash-sanitizer-daily/tiflash/contrib/googletest/googletest/src/gtest.cc:2811:28 (gtests_dbms+0xb80118d)
[2023-06-07T18:28:35.573Z]     #16 testing::internal::UnitTestImpl::RunAllTests() /home/jenkins/agent/workspace/tiflash-sanitizer-daily/tiflash/contrib/googletest/googletest/src/gtest.cc:5177:43 (gtests_dbms+0xb811340)
[2023-06-07T18:28:35.573Z]     #17 bool testing::internal::HandleSehExceptionsInMethodIfSupported<testing::internal::UnitTestImpl, bool>(testing::internal::UnitTestImpl*, bool (testing::internal::UnitTestImpl::*)(), char const*) /home/jenkins/agent/workspace/tiflash-sanitizer-daily/tiflash/contrib/googletest/googletest/src/gtest.cc:2443:10 (gtests_dbms+0xb826595)
[2023-06-07T18:28:35.573Z]     #18 bool testing::internal::HandleExceptionsInMethodIfSupported<testing::internal::UnitTestImpl, bool>(testing::internal::UnitTestImpl*, bool (testing::internal::UnitTestImpl::*)(), char const*) /home/jenkins/agent/workspace/tiflash-sanitizer-daily/tiflash/contrib/googletest/googletest/src/gtest.cc:2479:14 (gtests_dbms+0xb826595)
[2023-06-07T18:28:35.573Z]     #19 testing::UnitTest::Run() /home/jenkins/agent/workspace/tiflash-sanitizer-daily/tiflash/contrib/googletest/googletest/src/gtest.cc:4786:10 (gtests_dbms+0xb810628)
[2023-06-07T18:28:35.573Z]     #20 RUN_ALL_TESTS() /home/jenkins/agent/workspace/tiflash-sanitizer-daily/tiflash/contrib/googletest/googletest/include/gtest/gtest.h:2341:46 (gtests_dbms+0x5a5a771)
[2023-06-07T18:28:35.573Z]     #21 main /home/jenkins/agent/workspace/tiflash-sanitizer-daily/tiflash/dbms/src/TestUtils/gtests_dbms_main.cpp:117:16 (gtests_dbms+0x5a5a771)
[2023-06-07T18:28:35.573Z] 
[2023-06-07T18:28:35.573Z]   Thread T87 'cpu_pool' (tid=391249, running) created by main thread at:
[2023-06-07T18:28:35.573Z]     #0 pthread_create /llvm-project/compiler-rt/lib/tsan/rtl/tsan_interceptors_posix.cpp:977:3 (gtests_dbms+0x28198fd)
[2023-06-07T18:28:35.573Z]     #1 std::__1::__libcpp_thread_create(unsigned long*, void* (*)(void*), void*) /usr/local/bin/../include/c++/v1/__threading_support:514:10 (gtests_dbms+0xdec3880)
[2023-06-07T18:28:35.573Z]     #2 std::__1::thread::thread<void (DB::TaskThreadPool<DB::CPUImpl>::*)(unsigned long), DB::TaskThreadPool<DB::CPUImpl>*, unsigned long&, void>(void (DB::TaskThreadPool<DB::CPUImpl>::*&&)(unsigned long), DB::TaskThreadPool<DB::CPUImpl>*&&, unsigned long&) /usr/local/bin/../include/c++/v1/thread:307:16 (gtests_dbms+0xdec3880)
[2023-06-07T18:28:35.573Z]     #3 std::__1::thread* std::__1::construct_at<std::__1::thread, void (DB::TaskThreadPool<DB::CPUImpl>::*)(unsigned long), DB::TaskThreadPool<DB::CPUImpl>*, unsigned long&, std::__1::thread*>(std::__1::thread*, void (DB::TaskThreadPool<DB::CPUImpl>::*&&)(unsigned long), DB::TaskThreadPool<DB::CPUImpl>*&&, unsigned long&) /usr/local/bin/../include/c++/v1/__memory/construct_at.h:37:38 (gtests_dbms+0xdec0329)
[2023-06-07T18:28:35.573Z]     #4 void std::__1::allocator_traits<std::__1::allocator<std::__1::thread> >::construct<std::__1::thread, void (DB::TaskThreadPool<DB::CPUImpl>::*)(unsigned long), DB::TaskThreadPool<DB::CPUImpl>*, unsigned long&, void, void>(std::__1::allocator<std::__1::thread>&, std::__1::thread*, void (DB::TaskThreadPool<DB::CPUImpl>::*&&)(unsigned long), DB::TaskThreadPool<DB::CPUImpl>*&&, unsigned long&) /usr/local/bin/../include/c++/v1/__memory/allocator_traits.h:298:9 (gtests_dbms+0xdec0329)
[2023-06-07T18:28:35.573Z]     #5 void std::__1::vector<std::__1::thread, std::__1::allocator<std::__1::thread> >::__construct_one_at_end<void (DB::TaskThreadPool<DB::CPUImpl>::*)(unsigned long), DB::TaskThreadPool<DB::CPUImpl>*, unsigned long&>(void (DB::TaskThreadPool<DB::CPUImpl>::*&&)(unsigned long), DB::TaskThreadPool<DB::CPUImpl>*&&, unsigned long&) /usr/local/bin/../include/c++/v1/vector:948:5 (gtests_dbms+0xdec0329)
[2023-06-07T18:28:35.573Z]     #6 std::__1::thread& std::__1::vector<std::__1::thread, std::__1::allocator<std::__1::thread> >::emplace_back<void (DB::TaskThreadPool<DB::CPUImpl>::*)(unsigned long), DB::TaskThreadPool<DB::CPUImpl>*, unsigned long&>(void (DB::TaskThreadPool<DB::CPUImpl>::*&&)(unsigned long), DB::TaskThreadPool<DB::CPUImpl>*&&, unsigned long&) /usr/local/bin/../include/c++/v1/vector:1706:9 (gtests_dbms+0xdec0329)
[2023-06-07T18:28:35.573Z]     #7 DB::TaskThreadPool<DB::CPUImpl>::TaskThreadPool(DB::TaskScheduler&, DB::ThreadPoolConfig const&) /home/jenkins/agent/workspace/tiflash-sanitizer-daily/tiflash/dbms/src/Flash/Pipeline/Schedule/ThreadPool/TaskThreadPool.cpp:37:17 (gtests_dbms+0xdec0329)
[2023-06-07T18:28:35.573Z]     #8 DB::TaskScheduler::TaskScheduler(DB::TaskSchedulerConfig const&) /home/jenkins/agent/workspace/tiflash-sanitizer-daily/tiflash/dbms/src/Flash/Pipeline/Schedule/TaskScheduler.cpp:26:7 (gtests_dbms+0xde9d127)
[2023-06-07T18:28:35.573Z]     #9 std::__1::__unique_if<DB::TaskScheduler>::__unique_single std::__1::make_unique<DB::TaskScheduler, DB::TaskSchedulerConfig&>(DB::TaskSchedulerConfig&) /usr/local/bin/../include/c++/v1/__memory/unique_ptr.h:728:32 (gtests_dbms+0x5c0efe7)
[2023-06-07T18:28:35.573Z]     #10 DB::tests::ExecutorTest::SetUp() /home/jenkins/agent/workspace/tiflash-sanitizer-daily/tiflash/dbms/src/TestUtils/ExecutorTestUtils.cpp:65:31 (gtests_dbms+0x5c0efe7)
[2023-06-07T18:28:35.574Z]     #11 void testing::internal::HandleSehExceptionsInMethodIfSupported<testing::Test, void>(testing::Test*, void (testing::Test::*)(), char const*) /home/jenkins/agent/workspace/tiflash-sanitizer-daily/tiflash/contrib/googletest/googletest/src/gtest.cc:2443:10 (gtests_dbms+0xb8250f5)
[2023-06-07T18:28:35.574Z]     #12 void testing::internal::HandleExceptionsInMethodIfSupported<testing::Test, void>(testing::Test*, void (testing::Test::*)(), char const*) /home/jenkins/agent/workspace/tiflash-sanitizer-daily/tiflash/contrib/googletest/googletest/src/gtest.cc:2479:14 (gtests_dbms+0xb8250f5)
[2023-06-07T18:28:35.574Z]     #13 testing::Test::Run() /home/jenkins/agent/workspace/tiflash-sanitizer-daily/tiflash/contrib/googletest/googletest/src/gtest.cc:2513:3 (gtests_dbms+0xb7fe9a4)
[2023-06-07T18:28:35.574Z]     #14 testing::TestInfo::Run() /home/jenkins/agent/workspace/tiflash-sanitizer-daily/tiflash/contrib/googletest/googletest/src/gtest.cc:2693:11 (gtests_dbms+0xb8004c4)
[2023-06-07T18:28:35.574Z]     #15 testing::TestCase::Run() /home/jenkins/agent/workspace/tiflash-sanitizer-daily/tiflash/contrib/googletest/googletest/src/gtest.cc:2811:28 (gtests_dbms+0xb80118d)
[2023-06-07T18:28:35.574Z]     #16 testing::internal::UnitTestImpl::RunAllTests() /home/jenkins/agent/workspace/tiflash-sanitizer-daily/tiflash/contrib/googletest/googletest/src/gtest.cc:5177:43 (gtests_dbms+0xb811340)
[2023-06-07T18:28:35.574Z]     #17 bool testing::internal::HandleSehExceptionsInMethodIfSupported<testing::internal::UnitTestImpl, bool>(testing::internal::UnitTestImpl*, bool (testing::internal::UnitTestImpl::*)(), char const*) /home/jenkins/agent/workspace/tiflash-sanitizer-daily/tiflash/contrib/googletest/googletest/src/gtest.cc:2443:10 (gtests_dbms+0xb826595)
[2023-06-07T18:28:35.574Z]     #18 bool testing::internal::HandleExceptionsInMethodIfSupported<testing::internal::UnitTestImpl, bool>(testing::internal::UnitTestImpl*, bool (testing::internal::UnitTestImpl::*)(), char const*) /home/jenkins/agent/workspace/tiflash-sanitizer-daily/tiflash/contrib/googletest/googletest/src/gtest.cc:2479:14 (gtests_dbms+0xb826595)
[2023-06-07T18:28:35.574Z]     #19 testing::UnitTest::Run() /home/jenkins/agent/workspace/tiflash-sanitizer-daily/tiflash/contrib/googletest/googletest/src/gtest.cc:4786:10 (gtests_dbms+0xb810628)
[2023-06-07T18:28:35.574Z]     #20 RUN_ALL_TESTS() /home/jenkins/agent/workspace/tiflash-sanitizer-daily/tiflash/contrib/googletest/googletest/include/gtest/gtest.h:2341:46 (gtests_dbms+0x5a5a771)
[2023-06-07T18:28:35.574Z]     #21 main /home/jenkins/agent/workspace/tiflash-sanitizer-daily/tiflash/dbms/src/TestUtils/gtests_dbms_main.cpp:117:16 (gtests_dbms+0x5a5a771)
[2023-06-07T18:28:35.574Z] 
[2023-06-07T18:28:35.574Z] SUMMARY: ThreadSanitizer: data race /home/jenkins/agent/workspace/tiflash-sanitizer-daily/tiflash/contrib/client-c/third_party/libfiu/libfiu/fiu.c:212:11 in randd
[2023-06-07T18:28:35.574Z] ==================
```
- ComputeServerRunner.testErrorMessage
```
[2023-06-08T09:14:21.371Z] ==================
[2023-06-08T09:14:21.371Z] WARNING: ThreadSanitizer: data race (pid=430352)
[2023-06-08T09:14:21.371Z]   Read of size 8 at 0x7b58001cb498 by thread T2081:
[2023-06-08T09:14:21.371Z]     #0 DB::IProfilingBlockInputStream::collectCPUTimeNsImpl(bool) /home/jenkins/agent/workspace/tiflash-sanitizer-daily/tiflash/dbms/src/DataStreams/IProfilingBlockInputStream.cpp:423:52 (gtests_dbms+0xcbad8a8)
[2023-06-08T09:14:21.371Z]     #1 DB::IBlockInputStream::collectCPUTimeNs(bool) /home/jenkins/agent/workspace/tiflash-sanitizer-daily/tiflash/dbms/src/DataStreams/IBlockInputStream.cpp:118:12 (gtests_dbms+0xcba51c8)
[2023-06-08T09:14:21.371Z]     #2 DB::UnionBlockInputStream<(DB::StreamUnionMode)0, true>::collectCPUTimeNsImpl(bool)::'lambda'(DB::IBlockInputStream&)::operator()(DB::IBlockInputStream&) const /home/jenkins/agent/workspace/tiflash-sanitizer-daily/tiflash/dbms/src/DataStreams/UnionBlockInputStream.h:266:34 (gtests_dbms+0xdd85c14)
[2023-06-08T09:14:21.371Z]     #3 void DB::IBlockInputStream::forEachChild<DB::UnionBlockInputStream<(DB::StreamUnionMode)0, true>::collectCPUTimeNsImpl(bool)::'lambda'(DB::IBlockInputStream&)>(DB::UnionBlockInputStream<(DB::StreamUnionMode)0, true>::collectCPUTimeNsImpl(bool)::'lambda'(DB::IBlockInputStream&)&&) /home/jenkins/agent/workspace/tiflash-sanitizer-daily/tiflash/dbms/src/DataStreams/IBlockInputStream.h:164:17 (gtests_dbms+0xdd85c14)
[2023-06-08T09:14:21.371Z]     #4 DB::UnionBlockInputStream<(DB::StreamUnionMode)0, true>::collectCPUTimeNsImpl(bool) /home/jenkins/agent/workspace/tiflash-sanitizer-daily/tiflash/dbms/src/DataStreams/UnionBlockInputStream.h:264:9 (gtests_dbms+0xdd85c14)
[2023-06-08T09:14:21.371Z]     #5 DB::IBlockInputStream::collectCPUTimeNs(bool) /home/jenkins/agent/workspace/tiflash-sanitizer-daily/tiflash/dbms/src/DataStreams/IBlockInputStream.cpp:118:12 (gtests_dbms+0xcba51c8)
[2023-06-08T09:14:21.371Z]     #6 DB::IBlockInputStream::estimateCPUTimeNs() /home/jenkins/agent/workspace/tiflash-sanitizer-daily/tiflash/dbms/src/DataStreams/IBlockInputStream.h:123:16 (gtests_dbms+0xde6ed97)
[2023-06-08T09:14:21.371Z]     #7 DB::DataStreamExecutor::collectRequestUnit() /home/jenkins/agent/workspace/tiflash-sanitizer-daily/tiflash/dbms/src/Flash/Executor/DataStreamExecutor.cpp:85:41 (gtests_dbms+0xde6ed97)
[2023-06-08T09:14:21.371Z]     #8 DB::MPPTask::runImpl() /home/jenkins/agent/workspace/tiflash-sanitizer-daily/tiflash/dbms/src/Flash/Mpp/MPPTask.cpp:454:46 (gtests_dbms+0xde08c12)
[2023-06-08T09:14:21.371Z]     #9 DB::MPPTask::run()::$_17::operator()() const /home/jenkins/agent/workspace/tiflash-sanitizer-daily/tiflash/dbms/src/Flash/Mpp/MPPTask.cpp:169:97 (gtests_dbms+0xde0b363)
[2023-06-08T09:14:21.371Z]     #10 decltype(static_cast<DB::MPPTask::run()::$_17&>(fp)()) std::__1::__invoke<DB::MPPTask::run()::$_17&>(DB::MPPTask::run()::$_17&) /usr/local/bin/../include/c++/v1/type_traits:3918:1 (gtests_dbms+0xde0b363)
[2023-06-08T09:14:21.371Z]     #11 void std::__1::__invoke_void_return_wrapper<void, true>::__call<DB::MPPTask::run()::$_17&>(DB::MPPTask::run()::$_17&) /usr/local/bin/../include/c++/v1/__functional/invoke.h:61:9 (gtests_dbms+0xde0b363)
[2023-06-08T09:14:21.371Z]     #12 std::__1::__function::__alloc_func<DB::MPPTask::run()::$_17, std::__1::allocator<DB::MPPTask::run()::$_17>, void ()>::operator()() /usr/local/bin/../include/c++/v1/__functional/function.h:171:16 (gtests_dbms+0xde0b363)
[2023-06-08T09:14:21.371Z]     #13 std::__1::__function::__func<DB::MPPTask::run()::$_17, std::__1::allocator<DB::MPPTask::run()::$_17>, void ()>::operator()() /usr/local/bin/../include/c++/v1/__functional/function.h:345:12 (gtests_dbms+0xde0b363)
[2023-06-08T09:14:21.371Z]     #14 std::__1::__function::__value_func<void ()>::operator()() const /usr/local/bin/../include/c++/v1/__functional/function.h:498:16 (gtests_dbms+0xe56048d)
[2023-06-08T09:14:21.371Z]     #15 std::__1::function<void ()>::operator()() const /usr/local/bin/../include/c++/v1/__functional/function.h:1175:12 (gtests_dbms+0xe56048d)
[2023-06-08T09:14:21.371Z]     #16 decltype(static_cast<std::__1::function<void ()> const&>(fp)()) std::__1::__invoke<std::__1::function<void ()> const&>(std::__1::function<void ()> const&) /usr/local/bin/../include/c++/v1/type_traits:3918:1 (gtests_dbms+0xe56048d)
[2023-06-08T09:14:21.372Z] [92m[2023/06/08 17:14:12.604 +08:00] [0m[1m[31m[ERROR] [0m[35m[Exception.cpp:84] [0m[31m["virtual DB::SharedQueryBlockInputStream::~SharedQueryBlockInputStream(): Code: 0, e.displayText() = DB::Exception: Receiver state: CANCELED, e.what() = DB::Exception, Stack trace:\n\n\n       0x2836acb\t__interceptor_backtrace [gtests_dbms+42166987]\n                \t/llvm-project/compiler-rt/lib/tsan/../sanitizer_common/sanitizer_common_interceptors.inc:4202\n       0xe55a710\tStackTrace::StackTrace() [gtests_dbms+240494352]\n                \tdbms/src/Common/StackTrace.cpp:23\n       0x289727f\tDB::Exception::Exception(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, int) [gtests_dbms+42562175]\n                \tdbms/src/Common/Exception.h:46\n       0xdd82d6d\tDB::SharedQueryBlockInputStream::waitThread() [gtests_dbms+232271213]\n                \tdbms/src/DataStreams/SharedQueryBlockInputStream.h:183\n       0xdd824de\tDB::SharedQueryBlockInputStream::readSuffix() [gtests_dbms+232269022]\n                \tdbms/src/DataStreams/SharedQueryBlockInputStream.h:86\n       0xdd82583\tDB::SharedQueryBlockInputStream::~SharedQueryBlockInputStream() [gtests_dbms+232269187]\n                \tdbms/src/DataStreams/SharedQueryBlockInputStream.h:54\n       0xdd81b39\tstd::__1::__shared_ptr_emplace<DB::SharedQueryBlockInputStream, std::__1::allocator<DB::SharedQueryBlockInputStream> >::__on_zero_shared() [gtests_dbms+232266553]\n                \t/usr/local/bin/../include/c++/v1/__memory/shared_ptr.h:313\n       0x2a8af6d\tDB::IBlockInputStream::~IBlockInputStream() [gtests_dbms+44609389]\n                \tdbms/src/DataStreams/IBlockInputStream.h:128\n       0x2b83dd6\tDB::IProfilingBlockInputStream::~IProfilingBlockInputStream() [gtests_dbms+45628886]\n                \tdbms/src/DataStreams/IProfilingBlockInputStream.h:48\n       0xcb99e1d\tDB::ExpressionBlockInputStream::~ExpressionBlockInputStream() [gtests_dbms+213491229]\n                \tdbms/src/DataStreams/ExpressionBlockInputStream.h:29\n       0xb64d3d9\tstd::__1::__shared_ptr_emplace<DB::ExpressionBlockInputStream, std::__1::allocator<DB::ExpressionBlockInputStream> >::__on_zero_shared() [gtests_dbms+191157209]\n                \t/usr/local/bin/../include/c++/v1/__memory/shared_ptr.h:313\n       0x2a8af6d\tDB::IBlockInputStream::~IBlockInputStream() [gtests_dbms+44609389]\n                \tdbms/src/DataStreams/IBlockInputStream.h:128\n       0x2b83dd6\tDB::IProfilingBlockInputStream::~IProfilingBlockInputStream() [gtests_dbms+45628886]\n                \tdbms/src/DataStreams/IProfilingBlockInputStream.h:48\n       0xcb99e1d\tDB::ExpressionBlockInputStream::~ExpressionBlockInputStream() [gtests_dbms+213491229]\n                \tdbms/src/DataStreams/ExpressionBlockInputStream.h:29\n       0xb64d3d9\tstd::__1::__shared_ptr_emplace<DB::ExpressionBlockInputStream, std::__1::allocator<DB::ExpressionBlockInputStream> >::__on_zero_shared() [gtests_dbms+191157209]\n                \t/usr/local/bin/../include/c++/v1/__memory/shared_ptr.h:313\n       0x2a8af6d\tDB::IBlockInputStream::~IBlockInputStream() [gtests_dbms+44609389]\n                \tdbms/src/DataStreams/IBlockInputStream.h:128\n       0x2b83dd6\tDB::IProfilingBlockInputStream::~IProfilingBlockInputStream() [gtests_dbms+45628886]\n                \tdbms/src/DataStreams/IProfilingBlockInputStream.h:48\n       0xdd776e1\tDB::ExchangeSenderBlockInputStream::~ExchangeSenderBlockInputStream() [gtests_dbms+232224481]\n                \tdbms/src/DataStreams/ExchangeSenderBlockInputStream.h:25\n       0xdd1e639\tstd::__1::__shared_ptr_emplace<DB::ExchangeSenderBlockInputStream, std::__1::allocator<DB::ExchangeSenderBlockInputStream> >::__on_zero_shared() [gtests_dbms+231859769]\n                \t/usr/local/bin/../include/c++/v1/__memory/shared_ptr.h:313\n       0x2a8af6d\tDB::IBlockInputStream::~IBlockInputStream() [gtests_dbms+44609389]\n                \tdbms/src/DataStreams/IBlockInputStream.h:128\n       0xdd85a06\tDB::UnionBlockInputStream<(DB::StreamUnionMode)0, true>::~UnionBlockInputStream() [gtests_dbms+232282630]\n                \tdbms/src/DataStreams/UnionBlockInputStream.h:135\n       0xdd8411e\tstd::__1::__shared_ptr_emplace<DB::UnionBlockInputStream<(DB::StreamUnionMode)0, true>, std::__1::allocator<DB::UnionBlockInputStream<(DB::StreamUnionMode)0, true> > >::__on_zero_shared() [gtests_dbms+232276254]\n                \t/usr/local/bin/../include/c++/v1/__memory/shared_ptr.h:313\n       0xd5d50a6\tDB::ProcessListElement::releaseQueryStreams() [gtests_dbms+224219302]\n                \tdbms/src/Interpreters/ProcessList.cpp:278\n       0xd5d46d9\tDB::ProcessListEntry::~ProcessListEntry() [gtests_dbms+224216793]\n                \tdbms/src/Interpreters/ProcessList.cpp:194\n       0xd5d9e3e\tstd::__1::__shared_ptr_emplace<DB::ProcessListEntry, std::__1::allocator<DB::ProcessListEntry> >::__on_zero_shared() [gtests_dbms+224239166]\n                \t/usr/local/bin/../include/c++/v1/__memory/shared_ptr.h:313\n       0x2b26630\tDB::DAGContext::~DAGContext() [gtests_dbms+45246000]\n                \tdbms/src/Flash/Coprocessor/DAGContext.h:127\n       0xde0414c\tDB::MPPTask::~MPPTask() [gtests_dbms+232800588]\n                \tdbms/src/Flash/Mpp/MPPTask.cpp:123\n       0xddfdf8c\tstd::__1::__shared_ptr_pointer<DB::MPPTask*, std::__1::shared_ptr<DB::MPPTask>::__shared_ptr_default_delete<DB::MPPTask, DB::MPPTask>, std::__1::allocator<DB::MPPTask> >::__on_zero_shared() [gtests_dbms+232775564]\n                \t/usr/local/bin/../include/c++/v1/__memory/shared_ptr.h:267\n       0xde0b2af\tstd::__1::__function::__func<DB::MPPTask::run()::$_17, std::__1::allocator<DB::MPPTask::run()::$_17>, void ()>::destroy() [gtests_dbms+232829615]\n                \t/usr/local/bin/../include/c++/v1/__functional/function.h:327\n       0xe5601c3\tvoid* std::__1::__thread_proxy<std::__1::tuple<std::__1::unique_ptr<std::__1::__thread_struct, std::__1::default_delete<std::__1::__thread_struct> >, std::__1::thread DB::ThreadFactory::newThread<std::__1::function<void ()> >(bool, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, std::__1::function<void ()>&&)::'lambda'(auto&&...)> >(void*) [gtests_dbms+240517571]\n                \t/usr/local/bin/../include/c++/v1/thread:293\n       0x2819b9d\t__tsan_thread_start_func [gtests_dbms+42048413]\n                \t/llvm-project/compiler-rt/lib/tsan/rtl/tsan_interceptors_posix.cpp:967\n  0x7f19d671cea5\tstart_thread [libpthread.so.0+32421]"] [0m[90m[source="MPP<query:<query_ts:0, local_query_id:9, server_id:0, start_ts:41>,task_id:1> Aggregation aggregation_1"] [thread_id=184][0m
[2023-06-08T09:14:21.372Z]     #17 std::__1::invoke_result<std::__1::function<void ()> const&>::type std::__1::invoke<std::__1::function<void ()> const&>(std::__1::function<void ()> const&) /usr/local/bin/../include/c++/v1/__functional/invoke.h:93:12 (gtests_dbms+0xe56048d)
[2023-06-08T09:14:21.372Z]     #18 auto std::__1::thread DB::ThreadFactory::newThread<std::__1::function<void ()> >(bool, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, std::__1::function<void ()>&&)::'lambda'(auto&&...)::operator()<>(auto&&...) const /home/jenkins/agent/workspace/tiflash-sanitizer-daily/tiflash/dbms/src/Common/ThreadFactory.h:46:20 (gtests_dbms+0xe56048d)
[2023-06-08T09:14:21.372Z]     #19 decltype(static_cast<std::__1::function<void ()>>(fp)()) std::__1::__invoke<std::__1::thread DB::ThreadFactory::newThread<std::__1::function<void ()> >(bool, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, std::__1::function<void ()>&&)::'lambda'(auto&&...)>(std::__1::function<void ()>&&) /usr/local/bin/../include/c++/v1/type_traits:3918:1 (gtests_dbms+0xe560117)
[2023-06-08T09:14:21.372Z]     #20 void std::__1::__thread_execute<std::__1::unique_ptr<std::__1::__thread_struct, std::__1::default_delete<std::__1::__thread_struct> >, std::__1::thread DB::ThreadFactory::newThread<std::__1::function<void ()> >(bool, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, std::__1::function<void ()>&&)::'lambda'(auto&&...)>(std::__1::tuple<std::__1::function<void ()>, std::__1::thread DB::ThreadFactory::newThread<std::__1::function<void ()> >(bool, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, std::__1::function<void ()>&&)::'lambda'(auto&&...)>&, std::__1::__tuple_indices<>) /usr/local/bin/../include/c++/v1/thread:280:5 (gtests_dbms+0xe560117)
[2023-06-08T09:14:21.372Z]     #21 void* std::__1::__thread_proxy<std::__1::tuple<std::__1::unique_ptr<std::__1::__thread_struct, std::__1::default_delete<std::__1::__thread_struct> >, std::__1::thread DB::ThreadFactory::newThread<std::__1::function<void ()> >(bool, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, std::__1::function<void ()>&&)::'lambda'(auto&&...)> >(void*) /usr/local/bin/../include/c++/v1/thread:291:5 (gtests_dbms+0xe560117)
[2023-06-08T09:14:21.372Z] 
[2023-06-08T09:14:21.372Z]   Previous write of size 8 at 0x7b58001cb498 by thread T2143:
[2023-06-08T09:14:21.372Z]     #0 DB::BlockStreamProfileInfo::updateExecutionTime(unsigned long) /home/jenkins/agent/workspace/tiflash-sanitizer-daily/tiflash/dbms/src/DataStreams/BlockStreamProfileInfo.h:67:60 (gtests_dbms+0xcbac7a5)
[2023-06-08T09:14:21.372Z]     #1 DB::IProfilingBlockInputStream::readPrefix() /home/jenkins/agent/workspace/tiflash-sanitizer-daily/tiflash/dbms/src/DataStreams/IProfilingBlockInputStream.cpp:127:10 (gtests_dbms+0xcbac7a5)
[2023-06-08T09:14:21.372Z]     #2 DB::ParallelInputsProcessor<DB::UnionBlockInputStream<(DB::StreamUnionMode)0, true>::Handler, (DB::StreamUnionMode)0>::loop(unsigned long, DB::ParallelInputsProcessor<DB::UnionBlockInputStream<(DB::StreamUnionMode)0, true>::Handler, (DB::StreamUnionMode)0>::WorkingInputs&) /home/jenkins/agent/workspace/tiflash-sanitizer-daily/tiflash/dbms/src/DataStreams/ParallelInputsProcessor.h:297:23 (gtests_dbms+0xdd887c8)
[2023-06-08T09:14:21.372Z]     #3 DB::ParallelInputsProcessor<DB::UnionBlockInputStream<(DB::StreamUnionMode)0, true>::Handler, (DB::StreamUnionMode)0>::work(unsigned long, DB::ParallelInputsProcessor<DB::UnionBlockInputStream<(DB::StreamUnionMode)0, true>::Handler, (DB::StreamUnionMode)0>::WorkingInputs&) /home/jenkins/agent/workspace/tiflash-sanitizer-daily/tiflash/dbms/src/DataStreams/ParallelInputsProcessor.h:270:13 (gtests_dbms+0xdd88626)
[2023-06-08T09:14:21.372Z]     #4 DB::ParallelInputsProcessor<DB::UnionBlockInputStream<(DB::StreamUnionMode)0, true>::Handler, (DB::StreamUnionMode)0>::thread(unsigned long) /home/jenkins/agent/workspace/tiflash-sanitizer-daily/tiflash/dbms/src/DataStreams/ParallelInputsProcessor.h:253:9 (gtests_dbms+0xdd884fe)
[2023-06-08T09:14:21.372Z]     #5 DB::ParallelInputsProcessor<DB::UnionBlockInputStream<(DB::StreamUnionMode)0, true>::Handler, (DB::StreamUnionMode)0>::process()::'lambda'()::operator()() const /home/jenkins/agent/workspace/tiflash-sanitizer-daily/tiflash/dbms/src/DataStreams/ParallelInputsProcessor.h:128:81 (gtests_dbms+0xdd884fe)
[2023-06-08T09:14:21.373Z]     #6 decltype(static_cast<DB::ParallelInputsProcessor<DB::UnionBlockInputStream<(DB::StreamUnionMode)0, true>::Handler, (DB::StreamUnionMode)0>::process()::'lambda'()&>(fp)()) std::__1::__invoke<DB::ParallelInputsProcessor<DB::UnionBlockInputStream<(DB::StreamUnionMode)0, true>::Handler, (DB::StreamUnionMode)0>::process()::'lambda'()&>(DB::ParallelInputsProcessor<DB::UnionBlockInputStream<(DB::StreamUnionMode)0, true>::Handler, (DB::StreamUnionMode)0>::process()::'lambda'()&) /usr/local/bin/../include/c++/v1/type_traits:3918:1 (gtests_dbms+0xdd884fe)
[2023-06-08T09:14:21.373Z]     #7 void std::__1::__invoke_void_return_wrapper<void, true>::__call<DB::ParallelInputsProcessor<DB::UnionBlockInputStream<(DB::StreamUnionMode)0, true>::Handler, (DB::StreamUnionMode)0>::process()::'lambda'()&>(DB::ParallelInputsProcessor<DB::UnionBlockInputStream<(DB::StreamUnionMode)0, true>::Handler, (DB::StreamUnionMode)0>::process()::'lambda'()&) /usr/local/bin/../include/c++/v1/__functional/invoke.h:61:9 (gtests_dbms+0xdd884fe)
[2023-06-08T09:14:21.373Z]     #8 std::__1::__function::__alloc_func<DB::ParallelInputsProcessor<DB::UnionBlockInputStream<(DB::StreamUnionMode)0, true>::Handler, (DB::StreamUnionMode)0>::process()::'lambda'(), std::__1::allocator<DB::ParallelInputsProcessor<DB::UnionBlockInputStream<(DB::StreamUnionMode)0, true>::Handler, (DB::StreamUnionMode)0>::process()::'lambda'()>, void ()>::operator()() /usr/local/bin/../include/c++/v1/__functional/function.h:171:16 (gtests_dbms+0xdd884fe)
[2023-06-08T09:14:21.373Z]     #9 std::__1::__function::__func<DB::ParallelInputsProcessor<DB::UnionBlockInputStream<(DB::StreamUnionMode)0, true>::Handler, (DB::StreamUnionMode)0>::process()::'lambda'(), std::__1::allocator<DB::ParallelInputsProcessor<DB::UnionBlockInputStream<(DB::StreamUnionMode)0, true>::Handler, (DB::StreamUnionMode)0>::process()::'lambda'()>, void ()>::operator()() /usr/local/bin/../include/c++/v1/__functional/function.h:345:12 (gtests_dbms+0xdd884fe)
```

--- a/dbms/src/Common/FailPoint.cpp
+++ b/dbms/src/Common/FailPoint.cpp
@@ -49,6 +49,7 @@ namespace DB
     M(exception_during_mpp_register_tunnel_for_non_root_mpp_task) \
     M(exception_before_mpp_non_root_task_run)                     \
     M(exception_during_mpp_non_root_task_run)                     \
+    M(exception_during_query_run)                                 \
     M(exception_before_mpp_register_root_mpp_task)                \
     M(exception_before_mpp_register_tunnel_for_root_mpp_task)     \
     M(exception_before_mpp_root_task_run)                         \

--- a/dbms/src/Common/FailPoint.h
+++ b/dbms/src/Common/FailPoint.h
@@ -39,12 +39,17 @@ namespace ErrorCodes
 extern const int FAIL_POINT_ERROR;
 };
 
+#ifndef NDEBUG
 /// Macros to set failpoints.
 // When `fail_point` is enabled, throw an exception
 #define FAIL_POINT_TRIGGER_EXCEPTION(fail_point) \
     fiu_do_on(fail_point, throw Exception("Fail point " #fail_point " is triggered.", ErrorCodes::FAIL_POINT_ERROR);)
 // When `fail_point` is enabled, wait till it is disabled
 #define FAIL_POINT_PAUSE(fail_point) fiu_do_on(fail_point, FailPointHelper::wait(fail_point);)
+#else
+#define FAIL_POINT_TRIGGER_EXCEPTION(fail_point) ;
+#define FAIL_POINT_PAUSE(fail_point) ;
+#endif
 
 class FailPointChannel;
 class FailPointHelper

--- a/dbms/src/DataStreams/ExchangeSenderBlockInputStream.cpp
+++ b/dbms/src/DataStreams/ExchangeSenderBlockInputStream.cpp
@@ -12,31 +12,12 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-#include <Common/FailPoint.h>
 #include <DataStreams/ExchangeSenderBlockInputStream.h>
-#include <Flash/Coprocessor/DAGContext.h>
 
 namespace DB
 {
-namespace FailPoints
-{
-extern const char hang_in_execution[];
-extern const char exception_during_mpp_non_root_task_run[];
-extern const char exception_during_mpp_root_task_run[];
-} // namespace FailPoints
-
 Block ExchangeSenderBlockInputStream::readImpl()
 {
-    FAIL_POINT_PAUSE(FailPoints::hang_in_execution);
-    if (writer->dagContext().isRootMPPTask())
-    {
-        FAIL_POINT_TRIGGER_EXCEPTION(FailPoints::exception_during_mpp_root_task_run);
-    }
-    else
-    {
-        FAIL_POINT_TRIGGER_EXCEPTION(FailPoints::exception_during_mpp_non_root_task_run);
-    }
-
     Block block = children.back()->read();
     if (block)
     {

--- a/dbms/src/DataStreams/IProfilingBlockInputStream.cpp
+++ b/dbms/src/DataStreams/IProfilingBlockInputStream.cpp
@@ -12,13 +12,17 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+#include <Common/FailPoint.h>
 #include <DataStreams/IProfilingBlockInputStream.h>
 #include <Interpreters/ProcessList.h>
 #include <Interpreters/Quota.h>
 
-
 namespace DB
 {
+namespace FailPoints
+{
+extern const char exception_during_query_run[];
+} // namespace FailPoints
 
 namespace ErrorCodes
 {
@@ -45,6 +49,8 @@ Block IProfilingBlockInputStream::read()
 
 Block IProfilingBlockInputStream::read(FilterPtr & res_filter, bool return_filter)
 {
+    FAIL_POINT_TRIGGER_EXCEPTION(FailPoints::exception_during_query_run);
+
     if (total_rows_approx)
     {
         progressImpl(Progress(0, 0, total_rows_approx));

--- a/dbms/src/Flash/Coprocessor/DAGResponseWriter.h
+++ b/dbms/src/Flash/Coprocessor/DAGResponseWriter.h
@@ -42,7 +42,6 @@ public:
     /// flush cached blocks for batch writer
     virtual void flush() = 0;
     virtual ~DAGResponseWriter() = default;
-    const DAGContext & dagContext() const { return dag_context; }
 
 protected:
     Int64 records_per_chunk;

--- a/dbms/src/Flash/Executor/DataStreamExecutor.cpp
+++ b/dbms/src/Flash/Executor/DataStreamExecutor.cpp
@@ -76,6 +76,9 @@ int DataStreamExecutor::estimateNewThreadCount()
     return estimate_thread_cnt;
 }
 
+// TODO fix the data race on BlockStreamProfileInfo when the stream throws an error
+// and remove `race:dbms/src/DataStreams/BlockStreamProfileInfo.h` in tests/sanitize/tsan.suppression.
+// https://github.com/pingcap/tiflash/issues/7631
 RU DataStreamExecutor::collectRequestUnit()
 {
     // The cputime returned by BlockInputSrream is a count of the execution time of each thread.

--- a/dbms/src/Flash/Executor/QueryExecutor.cpp
+++ b/dbms/src/Flash/Executor/QueryExecutor.cpp
@@ -12,18 +12,26 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+#include <Common/FailPoint.h>
 #include <Flash/Executor/QueryExecutor.h>
 #include <Interpreters/Context.h>
 
 namespace DB
 {
+namespace FailPoints
+{
+extern const char hang_in_execution[];
+} // namespace FailPoints
+
 ExecutionResult QueryExecutor::execute()
 {
+    FAIL_POINT_PAUSE(FailPoints::hang_in_execution);
     return execute(ResultHandler{});
 }
 
 ExecutionResult QueryExecutor::execute(ResultHandler::Handler handler)
 {
+    FAIL_POINT_PAUSE(FailPoints::hang_in_execution);
     return execute(ResultHandler{handler});
 }
 

--- a/dbms/src/Flash/Mpp/MPPHandler.cpp
+++ b/dbms/src/Flash/Mpp/MPPHandler.cpp
@@ -50,14 +50,6 @@ void addRetryRegion(const ContextPtr & context, mpp::DispatchTaskResponse * resp
         }
     }
 }
-
-void injectFailPointBeforeMPPTaskRun(bool is_root_task)
-{
-    if (is_root_task)
-        FAIL_POINT_TRIGGER_EXCEPTION(FailPoints::exception_before_mpp_root_task_run);
-    else
-        FAIL_POINT_TRIGGER_EXCEPTION(FailPoints::exception_before_mpp_non_root_task_run);
-}
 } // namespace
 
 void MPPHandler::handleError(const MPPTaskPtr & task, String error)
@@ -89,7 +81,13 @@ grpc::Status MPPHandler::execute(const ContextPtr & context, mpp::DispatchTaskRe
         task->prepare(task_request);
 
         addRetryRegion(context, response);
-        injectFailPointBeforeMPPTaskRun(task->isRootMPPTask());
+
+#ifndef NDEBUG
+        if (task->isRootMPPTask())
+            FAIL_POINT_TRIGGER_EXCEPTION(FailPoints::exception_before_mpp_root_task_run);
+        else
+            FAIL_POINT_TRIGGER_EXCEPTION(FailPoints::exception_before_mpp_non_root_task_run);
+#endif
 
         task->run();
         LOG_INFO(log, "processing dispatch is over; the time cost is {} ms", stopwatch.elapsedMilliseconds());

--- a/dbms/src/Flash/Mpp/MPPTask.cpp
+++ b/dbms/src/Flash/Mpp/MPPTask.cpp
@@ -50,6 +50,8 @@ extern const char exception_before_mpp_register_tunnel_for_non_root_mpp_task[];
 extern const char exception_before_mpp_register_tunnel_for_root_mpp_task[];
 extern const char exception_during_mpp_register_tunnel_for_non_root_mpp_task[];
 extern const char force_no_local_region_for_mpp_task[];
+extern const char exception_during_mpp_non_root_task_run[];
+extern const char exception_during_mpp_root_task_run[];
 } // namespace FailPoints
 
 namespace
@@ -433,6 +435,17 @@ void MPPTask::runImpl()
             throw Exception("task not in running state, may be cancelled");
         }
         mpp_task_statistics.start();
+
+#ifndef NDEBUG
+        if (isRootMPPTask())
+        {
+            FAIL_POINT_TRIGGER_EXCEPTION(FailPoints::exception_during_mpp_root_task_run);
+        }
+        else
+        {
+            FAIL_POINT_TRIGGER_EXCEPTION(FailPoints::exception_during_mpp_non_root_task_run);
+        }
+#endif
 
         auto result = query_executor_holder->execute();
         LOG_INFO(log, "mpp task finish execute, success: {}", result.is_success);

--- a/dbms/src/Flash/Pipeline/Schedule/Tasks/EventTask.cpp
+++ b/dbms/src/Flash/Pipeline/Schedule/Tasks/EventTask.cpp
@@ -23,6 +23,7 @@ namespace FailPoints
 {
 extern const char random_pipeline_model_task_run_failpoint[];
 extern const char random_pipeline_model_cancel_failpoint[];
+extern const char exception_during_query_run[];
 } // namespace FailPoints
 
 #define EXECUTE(function)                                                                   \
@@ -33,6 +34,7 @@ extern const char random_pipeline_model_cancel_failpoint[];
     {                                                                                       \
         auto status = (function());                                                         \
         FAIL_POINT_TRIGGER_EXCEPTION(FailPoints::random_pipeline_model_task_run_failpoint); \
+        FAIL_POINT_TRIGGER_EXCEPTION(FailPoints::exception_during_query_run);               \
         return status;                                                                      \
     }                                                                                       \
     catch (...)                                                                             \

--- a/dbms/src/Flash/tests/gtest_compute_server.cpp
+++ b/dbms/src/Flash/tests/gtest_compute_server.cpp
@@ -30,7 +30,9 @@ extern const char exception_before_mpp_non_root_task_run[];
 extern const char exception_before_mpp_root_task_run[];
 extern const char exception_during_mpp_non_root_task_run[];
 extern const char exception_during_mpp_root_task_run[];
+extern const char exception_during_query_run[];
 } // namespace FailPoints
+
 namespace tests
 {
 LoggerPtr MPPTaskTestUtils::log_ptr = nullptr;
@@ -810,6 +812,7 @@ try
         FailPoints::exception_before_mpp_root_task_run,
         FailPoints::exception_during_mpp_non_root_task_run,
         FailPoints::exception_during_mpp_root_task_run,
+        FailPoints::exception_during_query_run,
     };
     size_t query_index = 0;
     for (const auto & failpoint : failpoint_names)

--- a/dbms/src/Flash/tests/gtest_compute_server.cpp
+++ b/dbms/src/Flash/tests/gtest_compute_server.cpp
@@ -797,6 +797,7 @@ CATCH
 TEST_F(ComputeServerRunner, testErrorMessage)
 try
 {
+    WRAP_FOR_SERVER_TEST_BEGIN
     startServers(3);
     setCancelTest();
     std::vector<String> failpoint_names{
@@ -841,6 +842,7 @@ try
             GTEST_FAIL();
         }
     }
+    WRAP_FOR_SERVER_TEST_END
 }
 CATCH
 

--- a/dbms/src/Operators/ExchangeSenderSinkOp.cpp
+++ b/dbms/src/Operators/ExchangeSenderSinkOp.cpp
@@ -12,19 +12,10 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-#include <Common/FailPoint.h>
-#include <Flash/Coprocessor/DAGContext.h>
 #include <Operators/ExchangeSenderSinkOp.h>
 
 namespace DB
 {
-namespace FailPoints
-{
-extern const char hang_in_execution[];
-extern const char exception_during_mpp_non_root_task_run[];
-extern const char exception_during_mpp_root_task_run[];
-} // namespace FailPoints
-
 void ExchangeSenderSinkOp::operatePrefix()
 {
     writer->prepare(getHeader());
@@ -37,18 +28,6 @@ void ExchangeSenderSinkOp::operateSuffix()
 
 OperatorStatus ExchangeSenderSinkOp::writeImpl(Block && block)
 {
-#ifndef NDEBUG
-    FAIL_POINT_PAUSE(FailPoints::hang_in_execution);
-    if (writer->dagContext().isRootMPPTask())
-    {
-        FAIL_POINT_TRIGGER_EXCEPTION(FailPoints::exception_during_mpp_root_task_run);
-    }
-    else
-    {
-        FAIL_POINT_TRIGGER_EXCEPTION(FailPoints::exception_during_mpp_non_root_task_run);
-    }
-#endif
-
     if (!block)
     {
         writer->flush();

--- a/tests/fullstack-test/mpp/mpp_fail.test
+++ b/tests/fullstack-test/mpp/mpp_fail.test
@@ -115,6 +115,12 @@ mysql> use test; set @@tidb_isolation_read_engines='tiflash'; set @@tidb_allow_m
 {#REGEXP}.*Fail point FailPoints::exception_during_mpp_root_task_run is triggered.*
 => DBGInvoke __disable_fail_point(exception_during_mpp_root_task_run)
 
+## exception during query run
+=> DBGInvoke __enable_fail_point(exception_during_query_run)
+mysql> use test; set @@tidb_isolation_read_engines='tiflash'; set @@tidb_allow_mpp=1; select count(value), id from t group by id;
+{#REGEXP}.*Fail point FailPoints::exception_during_query_run is triggered.*
+=> DBGInvoke __disable_fail_point(exception_during_query_run)
+
 ## exception during mpp hash build
 ## desc format='brief' select t1.id from test.t t1 join test.t t2 on t1.id = t2.id and t1.id <2 join (select id from test.t group by id) t3 on t2.id=t3.id;
 ## +-----------------------------------------+---------+-------------------+---------------+-------------------------------------------------------------------------+

--- a/tests/sanitize/tsan.suppression
+++ b/tests/sanitize/tsan.suppression
@@ -2,3 +2,4 @@ race:dbms/src/Common/TiFlashMetrics.h
 race:DB::Context::setCancelTest
 race:DB::getCurrentExceptionMessage
 race:fiu_fail
+race:dbms/src/DataStreams/BlockStreamProfileInfo.h

--- a/tests/sanitize/tsan.suppression
+++ b/tests/sanitize/tsan.suppression
@@ -1,3 +1,4 @@
 race:dbms/src/Common/TiFlashMetrics.h
 race:DB::Context::setCancelTest
 race:DB::getCurrentExceptionMessage
+race:fiu_fail

